bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/DanfossAirUnitHandler.java

```diff
@@ -0,0 +1,160 @@
+        super(thing);
+    }
+
+    private @Nullable ValueCache valueCache;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private @Nullable DanfossAirUnit hrv;
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateAllChannels();
+        } else {
+            try {
+                DanfossAirUnit danfossAirUnit = hrv;
+                Channel channel = Channel.getByName(channelUID.getIdWithoutGroup());
+                if (channel.getWriteAccessor() != null) {
+                    updateState(channelUID, channel.getWriteAccessor().access(danfossAirUnit, command));
+                }
+            } catch (IllegalArgumentException e) {
+                logger.info("{}", e.getMessage());
+            } catch (IOException ioe) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, ioe.getMessage());
+            } catch (Exception e) {
+                logger.info("Error updating channel {}: {}", channelUID, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        config = getConfigAs(DanfossAirUnitConfiguration.class);
+        valueCache = new ValueCache(config.updateUnchangedValuesEveryMillis);
+        try {
+            hrv = new DanfossAirUnit(InetAddress.getByName(config.host), 30046);
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Unknown host: " + config.host);
+        }
+
+        scheduler.execute(() -> {
+            try {
+                thing.setProperty(PROPERTY_UNIT_NAME, hrv.getUnitName());
+                thing.setProperty(PROPERTY_SERIAL, hrv.getUnitSerialNumber());
+                pollingJob = scheduler.scheduleWithFixedDelay(this::updateAllChannels, 5, config.refreshInterval,
+                        TimeUnit.SECONDS);
+                updateStatus(ThingStatus.ONLINE);
+
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            }
+        });
+    }
+
+    private void updateAllChannels() {
+        if (hrv == null) {
+            return;
+        }
+        DanfossAirUnit danfossAirUnit = hrv;
+        logger.debug("Updating DanfossHRV data '{}'", getThing().getUID());
+
+        try {
+            for (Channel channel : Channel.values()) {
+                if (Thread.interrupted()) {
+                    logger.debug("Polling thread interrupted...");
+                    return;
+                }
+                try {
+                    updateState(channel.getGroup().getGroupName(), channel.getChannelName(),
+                            channel.getReadAccessor().access(danfossAirUnit));
+                } catch (RuntimeException e) {
```

Comment:
You shouldn't catch RuntimeException here. If you are trying to catch exceptions that occur in your accessor then then your accessor should throw checked exceptions that you can then handle here.